pkg/hsperfdata/hsperfdata.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022-2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package hsperfdata
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+//HsperfdataPath returns the path of the hsperfdata file for a given pid,
+//it searches in all hsperfdata user directories, pids are assumed to be
+//unique regardless of username.
+//TODO(vthakkar): Add tests to makes sure above statement is true especially
+//with containers
```

Comment:
```suggestion
// HsperfdataPath returns the path of the hsperfdata file for a given pid,
// it searches in all hsperfdata user directories, pids are assumed to be
// unique regardless of username.
// TODO(vthakkar): Add tests to makes sure above statement is true especially
// with containers
```
🔧

Reply:
Yeah, it's a draft PR so no lint checks or code cleanups were done :D I generally do it after all the code works. But should probably change my editor setup, just so it's being done on save.